libafl_libfuzzer/libafl_libfuzzer_runtime/src/observers.rs

```diff
@@ -0,0 +1,366 @@
+use std::{
+    fmt::Debug,
+    hash::{Hash, Hasher},
+};
+
+use ahash::AHasher;
+use libafl::{
+    bolts::{tuples::Named, AsIter, HasLen},
+    executors::ExitKind,
+    inputs::UsesInput,
+    observers::{MapObserver, Observer, TimeObserver},
+    state::UsesState,
+    Error,
+};
+use num_traits::Bounded;
+use serde::{Deserialize, Serialize};
+
+static INITIAL_SIZE: usize = usize::MAX;
+static INITIAL_TIME: u64 = u64::MAX;
+
+pub trait ValueObserver: for<'de> Deserialize<'de> + Serialize + Debug + Named {
+    type ValueType: Bounded
+        + Default
+        + Copy
+        + Serialize
+        + for<'de> Deserialize<'de>
+        + PartialEq
+        + Hash
+        + Debug
+        + 'static;
+
+    fn value(&self) -> &Self::ValueType;
+
+    fn default_value(&self) -> &Self::ValueType;
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct MappedEdgeMapObserver<M, O> {
```

Comment:
SHouldn't some of these guys go to the main lib?

Reply:
We can change this in later PRs.